python-main.js

```diff
@@ -685,25 +747,19 @@ function web_editor(config) {
             $("#command-serial").attr("title", "Connect to your micro:bit via serial");
             $("#command-serial > .roundlabel").text("Open Serial");
             return;
-        } 
-
-        navigator.usb.requestDevice({
-            filters: [{vendorId: 0xD28}]
-        })
-        .then(function(device) {
+        }
 
+        // Check if we need to connect
+        if($("#command-connect").attr("title") == "Connect to your micro:bit"){
+            doConnect(undefined, true);
```

Comment:
Is it worth passing a custom event parameter rather than `undefined` here? 
We don't use it at the moment, but could the connection source be useful in the future?

Reply:
Possibly but I haven't for now but we might want to consider it if we need to detect multiple sources in the future.